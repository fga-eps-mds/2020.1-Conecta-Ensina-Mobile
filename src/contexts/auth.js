import AsyncStorage from '@react-native-community/async-storage';
import React, {createContext, useState, useEffect} from 'react';
import {Alert} from 'react-native';
import {HOST} from '@env';

export const AuthContext = createContext({});

export default function AuthProvider({children}) {
  const [user, setUser] = useState(null);
  const [typeUser, setTypeUser] = useState(null);
  const [teacher, setTeacher] = useState(null);
  const [student, setStudent] = useState(null);
  const [loading, setLoading] = useState(true);

  const Host = `${HOST}`;

  useEffect(() => {
    async function loadStorage() {
      const storageTypeUser = await AsyncStorage.getItem('Auth_typeUser');
      const storageUser = await AsyncStorage.getItem('Auth_user');

      if (storageTypeUser) {
        if (JSON.parse(storageTypeUser) === 1) {
          setTypeUser('Adm');
          setLoading(false);
        } else if (JSON.parse(storageTypeUser) === 2) {
          setTypeUser('Professor');
          setLoading(false);
        } else {
          setTypeUser('Aluno');
          setLoading(false);
        }
      }
      if (storageUser) {
        const dataUser = JSON.parse(storageUser);
        let usuario = {
          id: dataUser.id,
          firstName: dataUser.firstName,
          lastName: dataUser.lastName,
          email: dataUser.email,
          password: dataUser.password,
          role: dataUser.role,
          cellphone: dataUser.cellphone,
        };
        setUser(usuario);
        setLoading(false);
        if (dataUser.role == 2 || dataUser.role == 3) {
          const storageStudent = await AsyncStorage.getItem('Auth_student');
          if (storageStudent) {
            const dataStudent = JSON.parse(storageStudent);
            let estudante = {
              birthdate: dataStudent.birthdate,
              grade: dataStudent.grade,
              institution: dataStudent.school,
              cpf: dataStudent.cpf,
              cep: dataStudent.cep,
              number: dataStudent.number,
              details: dataStudent.details,
              description: dataStudent.description,
              special: dataStudent.special,
            };
            setStudent(estudante);
            setLoading(false);
          }
          if (dataUser.role == 2) {
            const storageTeacher = await AsyncStorage.getItem('Auth_teacher');
            if (storageTeacher) {
              const dataTeacher = JSON.parse(storageTeacher);
              let professor = {
                photo: dataTeacher.photo,
                video: dataTeacher.video,
                graduation_area: dataTeacher.graduation_area,
                degree: dataTeacher.degree,
                bank: dataTeacher.bank,
                agency: dataTeacher.agency,
                account: dataTeacher.account,
              };
              setTeacher(professor);
              setLoading(false);
            }
          }
        }
      }
      setLoading(false);
    }
    loadStorage();
  }, []);

  async function signIn(email, password) {
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: email,
        password: password,
      }),
    };
    const fetchResponse1 = await fetch(`${Host}/api/user/login`, settings);
    try {
      const data = await fetchResponse1.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Login efetuado com sucesso!') {
          console.log('login efetuado');
          storageTypeUser(data.role);
          renderData(data.id, data.role, password);
          if (data.role === 1) {
            setTypeUser('Adm');
            console.log('Adm');
          } else if (data.role === 2) {
            setTypeUser('Professor');
            console.log('Professor');
          } else {
            setTypeUser('Aluno');
            console.log('aluno');
          }
          setUser(data);
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
  } //close signIn function

  async function registerAluno(values, {setStatus}) {
    var ok = false;
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        password: values.password,
        cellphone: values.cellphone,
        birthdate: values.birthdate,
        grade: values.grade,
        institution: values.school,
        cpf: values.cpf,
        cep: values.cep,
        number: values.num,
        details: values.details,
        description: values.description,
        special: values.special,
      }),
    };
    const fetchResponse = await fetch(`${Host}/api/student/create`, settings);
    try {
      const data = await fetchResponse.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Estudante criado com sucesso!') {
          ok = true;

          renderData(
            data.data.user.id,
            data.data.user.role,
            data.data.user.password,
          );
          setTypeUser('Aluno');
          storageTypeUser(data.data.user.role);
        } else if (data.message.name) {
          if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.email
          ) {
            setStatus({email: 'Email j치 foi registrado'});
          } else if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.cpf
          ) {
            setStatus({cpf: 'CPF j치 foi registrado'});
          }
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
    if (ok) {
      RegistroAlert(3);
    }
  } //close registerAluno fuction

  async function registerProf(values, {setStatus}) {
    var ok = false;
    console.log(
      'firstName: ' +
        values.name +
        '\n' +
        'lastName: ' +
        values.surname +
        '\n' +
        'email: ' +
        values.email +
        '\n' +
        'password: ' +
        values.password +
        '\n' +
        'cellphone: ' +
        values.cellphone +
        '\n' +
        'birthdate: ' +
        values.birthdate +
        '\n' +
        'grade: ' +
        values.grade +
        '\n' +
        'institution: ' +
        values.school +
        '\n' +
        'cpf: ' +
        values.cpf +
        '\n' +
        'cep: ' +
        values.cep +
        '\n' +
        'number: ' +
        values.num +
        '\n' +
        'details: ' +
        values.details +
        '\n' +
        'description: ' +
        values.description +
        '\n' +
        'special: ' +
        values.special +
        '\n' +
        'photo: ' +
        values.photo +
        '\n' +
        'video: ' +
        values.video +
        '\n' +
        'graduation_area: ' +
        values.graduation_area +
        '\n' +
        'degree: ' +
        values.degree +
        '\n' +
        'bank: ' +
        values.bank +
        '\n' +
        'agency: ' +
        values.agency +
        '\n' +
        'account: ' +
        values.account +
        '\n',
    );
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        password: values.password,
        cellphone: values.cellphone,
        birthdate: values.birthdate,
        grade: values.grade,
        institution: values.school,
        cpf: values.cpf,
        cep: values.cep,
        number: values.num,
        details: values.details,
        description: values.description,
        special: values.special,
        photo: values.photo,
        video: values.video,
        graduation_area: values.graduation_area,
        degree: values.degree,
        bank: values.bank,
        agency: values.agency,
        account: values.account,
      }),
    };

    const fetchResponse = await fetch(`${Host}/api/teacher/create`, settings);
    try {
      const data = await fetchResponse.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Professor criado com sucesso!') {
          ok = true;
          renderData(
            data.data.user.id,
            data.data.user.role,
            data.data.user.password,
          );
          setTypeUser('Professor');
          storageTypeUser(data.data.user.role);
        } else if (data.message.name) {
          if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.email
          ) {
            setStatus({email: 'Email j치 foi registrado'});
          } else if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.cpf
          ) {
            setStatus({cpf: 'CPF j치 foi registrado'});
          }
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
    if (ok) {
      RegistroAlert(2);
    }
  } //close registerProf function

  const RegistroAlert = ({role}) => {
    Alert.alert(
      'Registro',
      'Registro Concluido com sucesso',
      [
        {
          text: 'Finalizar',
          onPress: () => {
            if (role === 2) {
              setTypeUser('Professor');
            } else {
              setTypeUser('Aluno');
            }
          },
        },
      ],
      {cancelable: false},
    );
  };

  async function renderData(id, role, password) {
    var ok = false;
    try {
      let response = await fetch(`${Host}/api/user/` + id);
      const data = await response.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      if (role === 2 || role === 3) {
        ok = true;
      }
      storageUser(usuario);
      setUser(usuario);
    } catch (error) {
      console.error(error);
    }
    if (ok) {
      try {
        let response = await fetch(`${Host}/api/student/` + id);
        const data = await response.json();
        let estudante = {
          birthdate: data.data.student.birthdate,
          grade: data.data.student.grade,
          institution: data.data.student.institution,
          cpf: data.data.student.cpf,
          cep: data.data.student.cep,
          number: data.data.student.number,
          details: data.data.student.details,
          description: data.data.student.description,
          special: data.data.student.special,
        };
        setStudent(estudante);
        storageStudent(estudante);
      } catch (error) {
        console.error(error);
      }
      if (role === 2) {
        try {
          let response = await fetch(`${Host}/api/teacher/` + id);
          const data = await response.json();
          let professor = {
            photo: data.data.teacher.photo,
            video: data.data.teacher.video,
            graduation_area: data.data.teacher.graduation_area,
            degree: data.data.teacher.degree,
            bank: data.data.teacher.bank,
            agency: data.data.teacher.agency,
            account: data.data.teacher.account,
          };
          setTeacher(professor);
          storageTeacher(professor);
        } catch (error) {
          console.error(error);
        }
      }
    }
  }

  async function updateUser(values, id) {
    const settings1 = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        cellphone: values.cellphone,
        password: user.password,
      }),
    };
    const fetchResponse1 = await fetch(`${Host}/api/user/${id}`, settings1);

    try {
      const data = await fetchResponse1.json();
      console.log('Success');
    } catch (error) {
      console.error('Error:', error);
    }
    if (user.role === 2 || user.role === 3) {
      const settings2 = {
        method: 'PUT',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          agentRole: user.role,
          birthdate: student.birthdate,
          grade: values.grade,
          institution: values.school,
          cpf: student.cpf,
          cep: values.cep,
          number: values.num,
          details: values.details,
          description: values.description,
          special: student.special,
        }),
      };
      const fetchResponse2 = await fetch(
        `${Host}/api/student/${id}`,
        settings2,
      );
      try {
        const data2 = await fetchResponse2.json();
        console.log('Success');
      } catch (error) {
        console.error('Error:', error);
      }
      if (user.role === 2) {
        const settings3 = {
          method: 'PUT',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            agentRole: user.role,
            video: values.video,
            graduation_area: teacher.graduation_area,
            degree: values.degree,
            bank: values.bank,
            agency: values.agency,
            account: values.account,
          }),
        };
        const fetchResponse3 = await fetch(
          `${Host}/api/teacher/${id}`,
          settings3,
        );
        try {
          const data3 = await fetchResponse3.json();
          console.log('Success');
        } catch (error) {
          console.error('Error:', error);
        }
      }
    }
    renderData(user.id, user.role, user.password);
  }

  async function storageTypeUser(data) {
    await AsyncStorage.setItem('Auth_typeUser', JSON.stringify(data));
  }

  async function storageUser(data) {
    await AsyncStorage.setItem('Auth_user', JSON.stringify(data));
  }

  async function storageStudent(data) {
    await AsyncStorage.setItem('Auth_student', JSON.stringify(data));
  }

  async function storageTeacher(data) {
    await AsyncStorage.setItem('Auth_teacher', JSON.stringify(data));
  }

  async function signOut() {
    await AsyncStorage.clear().then(() => {
      setUser(null);
      setTypeUser(null);
      setStudent(null);
      setTeacher(null);
    });
  }

  return (
    <AuthContext.Provider
      value={{
        signed: !!user,
        user,
        loading,
        teacher,
        student,
        typeUser,
        Host,
        signIn,
        registerAluno,
        registerProf,
        updateUser,
        signOut,
      }}>
      {children}
    </AuthContext.Provider>
  );
}
