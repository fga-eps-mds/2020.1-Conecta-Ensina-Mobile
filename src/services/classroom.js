export async function getNextClassroom(Host) {
  const fetchResponse = await fetch(
    `${Host}/api/classroom/nextClass/3bd7c190-ce64-4827-8c0c-58cfef45ad9f`,
  );
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function getUserClassroom(Host, id) {
  const fetchResponse = await fetch(`${Host}/api/classroom/userClasses/${id}`);
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function createClass(user, teacher, filter, student, Host) {
  const response = await fetch(`${Host}/api/classroom/create`, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      agentRole: user.role,
      teacher: teacher,
      student: user.id,
      grade: student.grade,
      subject: filter.subject,
      dtclass: filter.dtClass,
      duration: filter.duration,
      cep: student.cep,
      number: student.number,
      details: 'oi',
    }),
  });
  try {
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log(error);
  }

  console.log(response);
}
