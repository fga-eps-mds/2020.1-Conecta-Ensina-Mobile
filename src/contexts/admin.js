import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';

export const AdmContext = createContext({});

function AdmProvider({children}) {
  const [students, setStudents] = useState(null);
  const [userDB, setUserDB] = useState(null);
  const [teacher, setTeacher] = useState(null);

  const {Host, user} = useContext(AuthContext);

  async function getProfessorList() {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);

    console.log(Host);

    try {
      const data = await fetchResponse.json();
      setStudents(data.data.student);
    } catch (error) {
      return error;
    }
  }

  async function statusUpdate(id) {
    const settings = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        status: 1,
        agentRole: user.role,
      }),
    };
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    try {
      const data = await fetchResponse1.json();
      console.log('Success:', data);
    } catch (error) {
      console.error('Error:', error);
    }
  }

  async function getProfessoUser(id) {
    const fetchResponse = await fetch(`${Host}/api/user/${id}`);
    try {
      const data = await fetchResponse.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: data.data.user.password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      setUserDB(usuario);
      try {
        let response = await fetch(`${Host}/api/teacher/${id}`);
        const data = await response.json();
        let professor = {
          photo: data.data.teacher.photo,
          video: data.data.teacher.video,
          graduation_area: data.data.teacher.graduation_area,
          degree: data.data.teacher.degree,
          bank: data.data.teacher.bank,
          agency: data.data.teacher.agency,
          account: data.data.teacher.account,
        };
        setTeacher(professor);
      } catch (error) {
        console.error(error);
      }
    } catch (error) {
      return error;
    }
  }

  return (
    <AdmContext.Provider
      value={{
        students,
        userDB,
        teacher,
        getProfessorList,
        statusUpdate,
        getProfessoUser,
      }}>
      {children}
    </AdmContext.Provider>
  );
}

export default AdmProvider;
