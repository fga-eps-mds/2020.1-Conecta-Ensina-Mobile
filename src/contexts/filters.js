import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
//import * as Subj from '../services/subject';

export const FiltersContext = createContext({});

export default function FiltersProvider({children}) {
  const [filter, setFilter] = useState({
    subject: null,
    dtClass: null,
    duration: null,
    classType: null,
  });
  const {Host} = useContext(AuthContext);
  console.log(filter);

  function setSubjectFilter(subject) {
    let filters = {
      subject: subject,
      dtClass: filter.dtClass,
      duration: filter.duration,
      classType: filter.classType,
    };
    setFilter(filters);
  }

  function setDtClass(date) {
    let filters = {
      subject: filter.subject,
      dtClass: date,
      duration: filter.duration,
      classType: filter.classType,
    };
    setFilter(filters);
  }

  function setDurationFilters(duration) {
    let filters = {
      subject: filter.subject,
      dtClass: filter.dtClass,
      duration: duration,
      classType: filter.classType,
    };
    setFilter(filters);
  }

  function setClassType(type) {
    let filters = {
      subject: filter.subject,
      dtClass: filter.dtClass,
      duration: filter.duration,
      classType: type,
    };
    setFilter(filters);
  }

  return (
    <FiltersContext.Provider
      value={{
        filter,
        setSubjectFilter,
        setDtClass,
        setDurationFilters,
        setClassType,
      }}>
      {children}
    </FiltersContext.Provider>
  );
}
