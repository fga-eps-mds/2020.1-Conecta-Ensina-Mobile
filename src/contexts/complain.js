import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Student from '../services/student';

export const ComplainContext = createContext({});

export default function ComplainProvider({children}) {
  const [complain, setComplain] = useState({});
  const {Host, user} = useContext(AuthContext);

  async function createComplain(details,accused) {
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type':'application/json',
      },
      body:JSON.stringify({
        accused: accused,
        reported_by: user.id,
        details: details,
      })
    }
    const response = await fetch(`${Host}/api/complain/create`, settings)
    try{
      const data = response.json();
      console.log("Complain:", data);
      if (data.message === 'Den√∫ncia criada com sucesso!'){
        return data;
      }
    } catch(error) {
      console.error("error: ", error)
    }
  }

  return (
    <ComplainContext.Provider value={{complain, createComplain}}>
      {children}
    </ComplainContext.Provider>
  );
}