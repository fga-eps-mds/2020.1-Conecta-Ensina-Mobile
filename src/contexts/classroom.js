import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Class from '../services/classroom';
import * as Student from '../services/student';
import * as Address from '../services/findAddress';
import * as GeoCode from '../services/geocode';
import {FiltersContext} from '../contexts/filters';

export const ClassroomContext = createContext({});

export default function ClassroomProvider({children}) {
  const [classroom, setClassroom] = useState({});
  const [firstClass, setFirstClass] = useState({});
  const {Host, user} = useContext(AuthContext);
  const {filter} = useContext(FiltersContext);
  const [statusClasses, setStatusClasses] = useState([]);
  const [statusClass, setStatusClass] = useState({});
  const [coordenates, setCoordenates] = useState({});

  async function loadNextClass() {
    const response = await Class.getNextClassroom(Host);
    if (classroom !== response) {
      setFirstClass(response);
    }
  }
  async function createClass(teacher) {
    const student = await Student.getStudent(Host, user.id);
    const response = await Class.createClass(
      user,
      teacher,
      filter,
      student,
      Host,
    );
  }

  async function loadUserClasses() {
    const response = await Class.getUserClassroom(Host, user.id);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function getClassroom(status) {
    let teacher, student;
    if (user.role === 2) {
      student = '';
      teacher = user.id;
    } else {
      student = user.id;
      teacher = '';
    }
    const response = await Class.getClassroom(student, teacher, status, Host);
    if (classroom !== response) {
      console.log(response);
      setClassroom(response);
    }
    console.log(response.message);
  }

  async function updateStatusClassroom(id) {
    const response = await Class.getClass(Host, id);
    if (response.status === 1) {
      const response = await Class.updateStatusClassroom(id, 2, Host);
      console.log(response.message);
    } else if (response.status === 2) {
      const response = await Class.updateStatusClassroom(id, 3, Host);
      console.log(response.message);
    } else if (response.status === 3) {
      const response = await Class.updateStatusClassroom(id, 4, Host);
      console.log(response.message);
    } else if (response.status === 4) {
      const response = await Class.updateStatusClassroom(id, 5, Host);
      console.log(response.message);
    }
  }
  async function loadStatusClasses(status) {
    const response = await Class.getStatusClassroom(Host, user.id, status);
    if (classroom !== response) {
      console.log(response);
      setStatusClasses(response);
    }
    console.log(response);
  }

  async function readClass(id) {
    const response = await Class.getClass(Host, id);
    const address = await Address.findAddress(response.cep);
    let responseClass = {
      id: response.role,
      status: response.status,
      teacher: response.teacher,
      student: response.student,
      grade: response.grade,
      subject: response.subject,
      dtclass: response.dtClass,
      duration: response.duration,
      cep: response.cep,
      number: response.number,
      details: response.details,
      address: address,
    };
    if (classroom !== response) {
      setClassroom(responseClass);
    }
  }

  async function geoCode(cep) {
    const response = await GeoCode.geocode(cep);
    if (coordenates !== response) {
      console.log(response.results[0].geometry.location);
      setCoordenates(response.results[0].geometry.location);
    }
  }

  return (
    <ClassroomContext.Provider
      value={{
        classroom,
        loadNextClass,
        createClass,
        loadUserClasses,
        firstClass,
        getClassroom,
        updateStatusClassroom,
        loadStatusClasses,
        readClass,
        statusClasses,
        statusClass,
        setStatusClass,
        Host,
        geoCode,
        coordenates,
      }}>
      {children}
    </ClassroomContext.Provider>
  );
}
