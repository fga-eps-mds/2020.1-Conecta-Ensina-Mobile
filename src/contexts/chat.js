import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';
import * as Chat from '../services/chat';
import { ClassroomContext } from './classroom';

export const chatContext = createContext({});

export default function ChatProvider({children}) {
  const {Host, user} = useContext(AuthContext);
  const {classroom} = useContext(ClassroomContext);
  const [chat, setChat] = useState(null);

  useEffect(() => {
      readChat()
  }, [])

  async function createChat(text) {
    const response = await Chat.createMessage(
        user.id, 
        text, 
        classroom.id, 
        classroom.teacher, 
        classroom.student, 
        Host)
  }

  async function readChat() {
    console.log(classroom)
    const response = await Chat.getChat(Host, classroom.id);
    if (chat !== response) {
      setChat(response);
    }
  }
  return (
    <chatContext.Provider
      value={{
        readChat,
        createChat,
        chat
      }}>
      {children}
    </chatContext.Provider>
  );
}
