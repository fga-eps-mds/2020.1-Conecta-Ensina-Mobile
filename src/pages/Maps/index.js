import React, {useContext, useState, useEffect} from 'react';
import Theme from '../../../Theme';
import Background1 from '../../components/Background1';
import MapView from 'react-native-maps';
import {ClassroomContext} from '../../contexts/classroom';
export default function Maps({navigation}) {
  const {coordenates} = useContext(ClassroomContext);

  return (
    <Theme>
      <Background1>
        {coordenates && coordenates.lat && coordenates.lng && (
          <MapView
            style={{flex: 1}}
            initialRegion={{
              latitude: coordenates.lat,
              longitude: coordenates.lng,
              latitudeDelta: 0.0922,
              longitudeDelta: 0.0421,
            }}>
            <MapView.Marker
              coordinate={{
                latitude: coordenates.lat,
                longitude: coordenates.lng,
              }}
              title={'Local de aula'}
              description={'Local onde ocorrerÃ¡ a aula'}
            />
          </MapView>
        )}
      </Background1>
    </Theme>
  );
}
