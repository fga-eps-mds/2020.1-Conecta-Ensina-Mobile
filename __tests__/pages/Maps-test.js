import React from 'react';
import {render} from '@testing-library/react-native';
import Maps from '../../src/pages/Maps';
import {ClassroomContext} from '../../src/contexts/classroom';

describe('Testing Maps page', () => {
  jest.mock('react-native-maps', () => {
    const {View} = require('react-native');
    const MockMapView = (props: any) => {
      return <View>{props.children}</View>;
    };
    const MockMarker = (props: any) => {
      return <View>{props.children}</View>;
    };
    return {
      __esModule: true,
      default: MockMapView,
      Marker: MockMarker,
    };
  });

  const navigation = {
    navigate: jest.fn(),
  };

  let coordenates;
  beforeEach(() => {
    coordenates = {lat: -15.8644976, lng: -48.0351302};
  });

  test('Should match snapshot', () => {
    const tree = render(
      <ClassroomContext.Provider value={{coordenates}}>
        <Maps navigation={navigation} />
      </ClassroomContext.Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
