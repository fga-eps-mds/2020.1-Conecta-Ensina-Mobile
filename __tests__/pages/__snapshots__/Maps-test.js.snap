// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Maps Should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#184E79",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "row-reverse",
        "flexGrow": 22,
        "flexShrink": 1,
        "marginLeft": 30,
        "opacity": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 1000,
          "height": 58,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginLeft": 40,
          "opacity": 1,
          "width": 58,
        }
      }
      testID="userAvatar"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/user_blue.png",
          }
        }
        style={
          Array [
            Object {
              "height": 38,
              "width": 33,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 40,
            "marginRight": 35,
          },
        ]
      }
    >
      <Text
        medium={true}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 25,
              "textAlign": "center",
            },
          ]
        }
        white={true}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 78,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <AIRMap
      handlePanDrag={false}
      initialRegion={
        Object {
          "latitude": -15.8644976,
          "latitudeDelta": 0.0922,
          "longitude": -48.0351302,
          "longitudeDelta": 0.0421,
        }
      }
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      region={null}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <AIRMapMarker
        coordinate={
          Object {
            "latitude": -15.8644976,
            "longitude": -48.0351302,
          }
        }
        description="Local onde ocorrerÃ¡ a aula"
        onPress={[Function]}
        stopPropagation={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
            },
            undefined,
          ]
        }
        title="Local de aula"
      />
    </AIRMap>
  </View>
</View>
`;
